music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,397 @@
+            msg += f'the contextual inString ({inString}) refers to '
+            msg += 'measure number {measureNumber}. They should match.'
+            raise ValueError(msg)
+    else:  # inString and therefore start new line
+        inString = f'm{measureNumber}'
+
+    bt = intBeat(beat)
+
+    newString = f'{inString} b{bt} {chordString}'
+
+    return newString
+
+
+def intBeat(beat: Union[str, int, float, fractions.Fraction],
+            roundValue: int = 2):
+    '''
+    Converts beats to integers if possible, and otherwise to rounded decimals.
+    Accepts input as string, int, float, or fractions.Fraction.
+
+    >>> testInt = romanText.writeRoman.intBeat(1, roundValue=2)
+    >>> testInt
+    1
+
+    >>> testFrac = romanText.writeRoman.intBeat(8 / 3, roundValue=2)
+    >>> testFrac
+    2.67
+
+    >>> testStr = romanText.writeRoman.intBeat('0.666666666', roundValue=2)
+    >>> testStr
+    0.67
+
+    The roundValue sets the number of decimal places to round to. The default is two:
+
+    >>> testRound2 = romanText.writeRoman.intBeat(1.11111111, roundValue=2)
+    >>> testRound2
+    1.11
+
+    But this can be set to any integer:
+
+    >>> testRound1 = romanText.writeRoman.intBeat(1.11111111, roundValue=1)
+    >>> testRound1
+    1.1
+    '''
+
+    options = (str, int, float, fractions.Fraction)
+
+    if not isinstance(beat, options):
+        raise TypeError(f'Beat, (currently {beat}) must be one of {options}.')
+
+    if isinstance(beat, int):
+        return beat
+
+    if type(beat) in [str, fractions.Fraction]:
+        beat = float(beat)
+
+    # Now beat is definitely a float
+    if int(beat) == beat:
+        return int(beat)
+    else:
+        return round(beat, roundValue)
+
+
+# ------------------------------------------------------------------------------
+
+class Test(unittest.TestCase):
+    '''
+    Tests for two analysis cases (the smallest rntxt files in the music21 corpus)
+    along with two test by modifying those scores.
+
+    Additional tests for the stand alone functions rnString and intBeat and
+    error cases for all three.
+    '''
+
+    def testTwoCorpusPiecesAndTwoCorruptions(self):
+
+        from music21 import corpus
+        from music21 import meter
+
+        scoreBach = corpus.parse('bach/choraleAnalyses/riemenschneider004.rntxt')  # Smallest file
+
+        rnaBach = RnWriter(scoreBach)
+        self.assertEqual(rnaBach.combinedList[-1], 'm10 b1 V6/V b2 V b3 I')  # NB b1
+
+        # --------------------
+
+        scoreBach.parts[0].measure(3).insert(0, meter.TimeSignature('10/8'))
+        scoreBach.parts[0].measure(3).insert(1, meter.TimeSignature('5/8'))
+
+        wonkyBach = RnWriter(scoreBach)
+
+        self.assertEqual(wonkyBach.combinedList[9], 'Time Signature: 10/8')
+        self.assertEqual(wonkyBach.combinedList[10],
+                         'Note: further time signature change(s) unprocessed: [\'5/8\']')
```

Comment:
To address Myke's concern with it being brittle to index specific lines (in case analyst, proofreader, etc. supported later), wouldn't it just be sufficient to do:
```
self.assertIn('Note: further...', wonkyBach.combinedList)
```
You might need to add a trailing newline.

Reply:
Yes, and then it would also be possible to check that the TS Note is on the line after the Time Signature setting.  

Five years ago I would have accepted this, but since then I've learned more about what makes me have to return to the code later when I can't get a hold of the original writers, so I have gotten pickier over time.  :-)